adlfs/transfer.py

```diff
@@ -0,0 +1,312 @@
+# -*- coding: utf-8 -*-
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+High performance multi-threaded module to up/download
+
+Calls method in `core` with thread pool executor to ensure the network
+is used to its maximum throughput.
+
+Only implements upload and download of (massive) files and directory trees.
+"""
+from collections import namedtuple
+from concurrent.futures import ThreadPoolExecutor
+import logging
+import multiprocessing
+import os
+import pickle
+import threading
+import time
+import uuid
+
+logger = logging.getLogger(__name__)
+
+File = namedtuple('File', 'src dst state nbytes start stop chunks')
+Chunk = namedtuple('Chunk', 'name state offset retries')
+
+
+class DisjointState(object):
+    """
+    Collection of disjoint sets containing state about objects.
```

Comment:
This description doesn't tell me what this class is for. I suspect it is meant to be purely internal.